frontend/src/app/http-interceptor.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { HttpRequest, HttpInterceptor as AngularHttpInterceptor, HttpHandler, HttpEvent } from '@angular/common/http';
+import { Observable } from 'rxjs/Observable';
+import { environment } from '../environments/environment';
+
+@Injectable()
+export class HttpInterceptor implements AngularHttpInterceptor {
+
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    let authReq = req;
+    if (req.url.indexOf(environment.apiRoot) === 0) {
```

Comment:
:+1:

Reply:
ES6 provides new method .includes on String.prototype. Returns boolean value. Usefull in that case. 
https://developer.mozilla.org/pl/docs/Web/JavaScript/Reference/Global_Objects/String/includes

Based on that I would write:
if (req.url.includes(environment.apiRoot) ) {

Reply:
It's not exactly the same, as here we also check if it is at the beginning of url (as I understand `includes` checkes if it contains, but not where).

Reply:
We should use here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith

Reply:
True, my bad in uderstanding this line.
.includes is a replacement for string.indexOf('text') > -1

Reply:
I just  realized that we don't want to append auth headers to all requests. I will test if it works when user is not logged in.

Reply:
It's OK - no error is thrown when cookie does not exist.